Display set_nodelay to GDScript

schweigert opened the pull request:
Hi guys,

This made changes to issue #16187.

This PR displays the interface to disable Nagle Algorithm with GDScript.

Zephilinox commented:
Perhaps before exposing this to GDScript it could be considered renaming it to set_no_delay internally, since that seems like a typo.

Noshyaar commented on core/io/stream_peer_tcp.cpp:
```diff
@@ -55,6 +55,7 @@ void StreamPeerTCP::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_connected_host"), &StreamPeerTCP::get_connected_host);
 	ClassDB::bind_method(D_METHOD("get_connected_port"), &StreamPeerTCP::get_connected_port);
 	ClassDB::bind_method(D_METHOD("disconnect_from_host"), &StreamPeerTCP::disconnect_from_host);
+	ClassDB::bind_method(D_METHOD("set_nodelay"), &StreamPeerTCP::set_nodelay);
```
Should be `D_METHOD("set_nodelay", "enabled")`.

Thanks for contributing!

Schweigert commented:
@Zephilinox NODELAY is a flag of the TCP protocol to disable the Nagle Algorithm. The name is partially correct, so I prefer to keep the probable error.

Noshyaar commented:
One more thing, would you mind squash commits together? ([PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request))

Schweigert commented:
Yes, no problem. I'm used to working with squash commits.
@Zephilinox I renamed to set_no_delay.

akien-mga commented on doc/classes/StreamPeerTCP.xml:
```diff
@@ -29,6 +29,15 @@
+			<argument index="0" name="enabled" type="bool">
+			</argument>
+			<description>
+				Disable Naggle Algorithm.
```
I feel like some more details could be useful. I would need to google "Naggle algorithm" to understand what this method does, as I'm not a TCP expert. It's also not clear to untrained eyes how "no deplay" and "Naggle algorithm" are linked.

Schweigert replied:
Now it is better described and with a warning that disabling this, you will spend more bandwidth.

hpvb commented on doc/classes/StreamPeerTCP.xml:
```diff
@@ -29,6 +29,16 @@
+			<argument index="0" name="enabled" type="bool">
+			</argument>
+			<description>
+				Disables the Nagle algorithm to speed up the TCP protocol.
```
I don't think this is a correct description of what Nagle algorithm does. If you don't need TCP_NODELAY then disabling it might make your overall application slower. I'd much prefer something like "Disable Nagle algorithm to improve latency for small packets. Note that for applications that send large packets, or need to transfer a lot of data, this can reduce total bandwidth."

schweigert committed:
Display set_nodelay to GDScript

Pass enabled arg

Rename set_nodelay to set_no_delay

Add description to the method

Change description
